hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                If True, report feedback to stdout.
+        """
+        if keys is None:
+            keys = self.svals.keys()
+        if u is None:
+            u = self.umodes
+        if v is None:
+            v = self.vmodes
+
+        if not hasattr(self, 'pcomps'):
+            self.pcomps = DataContainer({})
+
+        for k in keys:
+            if k in self.pcomps and not overwrite:
+                echo("{} exists in pcomps and overwrite == False, skipping...".format(k), verbose=verbose)
+                continue
+
+            self.pcomps[k] = _form_PCs(u[k], v[k])
+        if hasattr(u, 'times'):
+            self.pcomps.times = u.times
+
+    def build_model(self, keys=None, Nkeep=None, overwrite=False, increment=False, verbose=True):
+        """
+        Sum principal components to get a model.
+
+        Sum principal components dotted with singular values and add to 
+        the pcomp_model.
+
+        Args:
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            Nkeep : int
+                Number of principal components to keep when forming model.
+            overwrite : bool
+                If True, overwrite output data if it exists.
+            increment : bool
+                If key already exists in pcomp_model, add the new model
+                to it, rather than overwrite it. This supercedes overwrite
+                if both are true.
+            verbose : bool
+                If True, report feedback to stdout.
+        """
+        # get keys
+        if keys is None:
+            keys = self.pcomps.keys()
+
+        # setup containers
+        if not hasattr(self, "pcomp_model"):
+            self.pcomp_model = DataContainer({})
+
+        # iterate over keys
+        for k in keys:
+            if k not in self.pcomps:
+                echo("{} not in pcomps, skipping...".format(k), verbose=verbose)
+                continue
+            if k in self.pcomp_model and (not overwrite and not increment):
+                echo("{} in pcomp_model and overwrite == increment == False, skipping...".format(k), verbose=verbose)
+                continue
+
+            # sum PCs
+            model_fft = sum_principal_components(self.svals[k], self.pcomps[k], Nkeep=Nkeep)
+            if k not in self.pcomp_model:
+                self.pcomp_model[k] = model_fft
+            elif k in self.pcomp_model and increment:
+                self.pcomp_model[k] += model_fft
+            else:
+                self.pcomp_model[k] = model_fft
+
+    def subtract_model(self, keys=None, data=None, overwrite=False, ifft=True, ifftshift=True,
+                       window='none', alpha=0.2, edgecut_low=0, edgecut_hi=0, verbose=True):
+        """
+        Subtract pcomp_model from data.
+
+        FFT pcomp_model to frequency space, divide by window, and subtract from data.
+        Inserts FFT of pcomp_model into self.pcomp_model_fft and 
```

Comment:
I find this nomenclature confusing. `model_fft` and `pcomp_model` are in delay space, but `pcomp_model_fft` is in frequency space. That's not intuitive at all.